nasa-gibs/worldview
# Historical events

## Description

Fixes #3403 
Fixes #3508 
Fixes #3428 

### TODO

- [x] Update test suite
- [x] Finish unit tests
- [x] Fix Firefox date validation issue
- [x] Update event modal text

### Summary of Changes

* Hard-code event categories in config file: `config/default/common/config/wv.json/naturalEvents.json` instead of requesting from EONET API
* Tweak sidebar modal styles for more consistency
* Add `<ModalFooter />` to modals if the `footer: true` option is sent in call to `toggleCustomContent()`.  Portals can be used to insert content like action buttons.
* Remove tabIndex attribute from `<DateInputColumn />`, instead use generated class names to move focus among sibling inputs instead.
* Created new `<DateRangeSelector />` component which serves as a start/end date selector. Now used in animation widget as well as event filter modal
* Simplify params for `areCoordinatesWithinExtent()` util function to only require state projection object and coords to check.
* Create standalone `<EventIcon />` component for showing event icon of given type with hover tooltip
* Convert `<FooterContent />` component to functional component

### Additional Notes

From the comments:
```
/**
 * Since the EONET API bounding box parameter only accepts a geographic coordinate format,
 * the bounding box extents we send during the API call may return events outside of the
 * visible extent of polar projections within the app. Therefore, we filter out any geometries
 * for an event that are outside the visible extent (to prevent track points from rendering beyond it)
 * and, we filter out any events that have no visible geometries in the current extent entirely.
 *
 * We also need to filter the categories on events so that only categories
 * which we support and, which are currently selected in the event filter remain on an 
 * event object so that an inappropriate icon does not show.  (E.g. event belongs to
 * "Severe Storms" and "Snow" but was returned from a search for "Snow" events only, we don't
 * want the "Severe Storms" icon to show)
 */
```

## How To Test

In general:

* Try various combinations of dates and categories with different projections
* For each set of results, check the URL params and copy/paste to a new tab to confirm proper functionality
* Verify date inputs for timeline, animation widget still behave as intended since these components were affected by changes in this PR
